src/shared/components/case-viewer/case-event-trigger/case-event-trigger.component.ts

```diff
@@ -74,7 +74,7 @@ export class CaseEventTriggerComponent implements OnInit, OnDestroy {
   submitted(event: any): void {
     let eventStatus: string = event['status'];
     this.router
-      .navigate([this.parentUrl])
+      .navigate([this.parentUrl], { queryParams: { 'onErrorCaseList': true } })
```

Comment:
any better way other than queryParam?

Reply:
mmm I'd avoid introducing a query param. Have you considered checking the previous url? 
https://stackoverflow.com/questions/41038970/how-to-determine-previous-page-url-in-angular

Reply:
well on case.resolver.ts i need to know whether it's a redirect to caseList page or standard error page. So, need to pass some flag/property to determine that.

Reply:
what I mean is, on `case.resolver.ts` we check:
 if `error.status === 404` and `coming from submit case url` then redirect to case list

Reply:
done.